File: common/src/main/java/scooterdelta/challenge/bot/common/local/FileState.java
package scooterdelta.challenge.bot.common.local;

import java.io.File;
import java.util.Objects;

public class FileState {

    private final File workingDirectory;
    private final String playerKey;

    public FileState(File workingDirectory, String playerKey) {
        this.workingDirectory = workingDirectory;
        this.playerKey = playerKey;
    }

    public File getWorkingDirectory() {
        return workingDirectory;
    }

    public String getPlayerKey() {
        return playerKey;
    }

    @Override
    public String toString() {
        return "FileState{" +
                "workingDirectory=" + workingDirectory +
                ", playerKey='" + playerKey + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        FileState fileState = (FileState) o;
        return Objects.equals(workingDirectory, fileState.workingDirectory) &&
                Objects.equals(playerKey, fileState.playerKey);
    }

    @Override
    public int hashCode() {
        return Objects.hash(workingDirectory, playerKey);
    }
}

File: common/src/main/kotlin/scooterdelta/challenge/bot/common/state/local/FileState.kt
package scooterdelta.challenge.bot.common.state.local

import java.io.File

class FileState (val workingDirectory : File, val playerKey : String) {

    override fun toString(): String {
        return "FileState(workingDirectory=$workingDirectory, playerKey='$playerKey')"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other?.javaClass != javaClass) return false

        other as FileState

        if (workingDirectory != other.workingDirectory) return false
        if (playerKey != other.playerKey) return false

        return true
    }

    override fun hashCode(): Int {
        var result = workingDirectory.hashCode()
        result = 31 * result + playerKey.hashCode()
        return result
    }
}